.github/workflows/teamcity-pr-checks.yml

```diff
@@ -0,0 +1,46 @@
+name: TeamCity Configuration Tests
+permissions: read-all
+
+on:
+  workflow_dispatch:
+  pull_request:
+    paths:
+      - '.github/workflows/teamcity-pr-checks.yml'
+      - 'mmv1/third_party/terraform/!.teamcity'
+
+
+jobs:
+  teamcity-config-tests:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+        with:
+          fetch-depth: 0
+
+      - name: Setup Java
+        uses: actions/setup-java@387ac29b308b003ca37ba93a6cab5eb57c8f5f93 # v4.0.0
+        with:
+          distribution: zulu
+          java-version: 17
+          java-package: jdk
+
+      - name: Cache Maven files
+        uses: actions/cache@13aacd865c20de90d75de3b17ebe84f7a17d57d2 # v4.0.0
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
+
+      - name: Download TeamCity Tools
+        run : |
+          cd mmv1/third_party/terraform/.teamcity
+          make tools
+
+      # Running the tests twice ensures that the tests are discovered and run
+      - name: Run TeamCity Tests
+        run : |
+          cd mmv1/third_party/terraform/.teamcity
+          make test
+          make test
```

Comment:
I don't know the exact reason why, but if you only run this command once the tests aren't detected and the test runner reports `[INFO] No tests to run.`:

![Screenshot 2024-02-09 at 12 13 42](https://github.com/GoogleCloudPlatform/magic-modules/assets/15078782/41069dd3-cfb6-46f6-8461-5dc32f1bbf3c)


Instead the logs should show something like this:

```
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running tests.NightlyTestProjectsTests
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.164 s - in tests.NightlyTestProjectsTests
[INFO] Running tests.BuildConfigurationFeatureTests
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.032 s - in tests.BuildConfigurationFeatureTests
[INFO] Running tests.ContextParameterHandlingTests
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.144 s - in tests.ContextParameterHandlingTests
[INFO] Running tests.VcsTests
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.003 s - in tests.VcsTests
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 7, Failures: 0, Errors: 0, Skipped: 0
```

Reply:
We can see the TESTS output in the GHA run for 1bf8b76 [in this line here](https://github.com/GoogleCloudPlatform/magic-modules/actions/runs/7843714618/job/21404633987?pr=9954#step:6:10766)

Reply:
I think the basis of this behaviour is the pom.xml file, [maybe this section](https://github.com/GoogleCloudPlatform/magic-modules/blob/b758bf574d9159b924694e9083f211b97a36de09/mmv1/third_party/terraform/.teamcity/pom.xml#L47-L88), but I'm really not an experienced Kotlin/Java developer. Given that the pom.xml file we use strongly resembles the Azure provider team's pom.xml file, who are the SMEs for TeamCity in HashiCorp, I'd like to avoid digging into this solo.

Reply:
Interesting, but it works fine locally, right?